kratos/includes/properties.h

```diff
@@ -277,30 +295,138 @@ class Properties : public IndexedObject
         return result_key;
     }
 
+    /**
+     * @brief This method returns the number of subproperties
+     * @return The current number of subproperties
+     */
+    std::size_t NumberOfSubproperties()
+    {
+        return mSubPropertiesList.size();
+    }
+
+    /**
+     * @brief This method insert a new property into the list of subproperties
+     * @param pNewSubProperty The new property to be added
+     */
+    void AddSubProperty(Properties::Pointer pNewSubProperty)
+    {
+        mSubPropertiesList.insert(mSubPropertiesList.begin(), pNewSubProperty);
```

Comment:
what happens if the subproperty already exists? (to be decided and documented...i would personally suggest throwing as a minimum a warning and ideally an error)

Reply:
There is a check for that:

https://github.com/KratosMultiphysics/Kratos/blob/80332ee8dec355c6d78e9d153df521c68f90d000/kratos/utilities/read_materials_utility.cpp#L239

Reply:
i would say that you need to check if it exists in the root modelpart , not in the current modelpart.

also by doing this you are implying that property Ids need to be unique within a modelpart tree. It is an option, 
but ... is it a decision?